protos/tcp/tcp.go

```diff
@@ -135,7 +162,7 @@ func (tcp *Tcp) Process(tcphdr *layers.TCP, pkt *protos.Packet) {
 			// create
 			stream = &TcpStream{id: tcp.getId(), tuple: &pkt.Tuple, protocol: protocol, tcp: tcp}
 			stream.tcptuple = common.TcpTupleFromIpPort(stream.tuple, stream.id)
-			tcp.streamsMap[pkt.Tuple.Hashable()] = stream
+			tcp.putStream(pkt.Tuple.Hashable(), stream)
```

Comment:
I think there's a theoretical race here: if two threads/goroutines do getStream at the same time and find that the stream doesn't yet exist, they can both call putStream() afterwards and overwrite each other's data. I think this cannot happen at the moment because all TCP packets are processed by a single goroutine anyway. Also, when we will scale to multiple threads, we're likely to want to send all packets from the same TCP stream to a single thread to avoid issues like this by design.

So I'm not saying that we necessarily have to change this, but we might want to be defensive here nevertheless. I'm curious what the others think as well.